docs/src/architecture/09_architecture_decisions/0011-align-be-fe-dev-releases.md

```diff
@@ -0,0 +1,132 @@
+
+4. **Breaking changes in main**: Large breaking changes (like API V2) merged to `main` cause problems
+for both teams - backend loses ability to release from `main`, and frontend can't deploy without breaking changes.
+
+We agreed (Oct 13, 2025 discussion) to standardize on **releasing from tags only (not every merge)** for both BE and FE,
+with **feature-branch environments** for testing larger breaking changes before they land in `main`.
+
+## Assumptions
+
+- **OpenAPI** is (and remains) our single source of truth for public HTTP APIs.
+- Teams accept **spec-first** changes: spec updates land in PRs **before** code changes.
+- CI/CD can:
+  - compute **breaking diffs** between the current spec and the last released spec,
+  - run **spec conformance tests** against a running provider.
+- We can operate **channelized promotions** (unstable → testing → stable) and **preview envs** per PR/feature branch.
+- Weekly **release train** (lightweight) is acceptable; hotfixes remain possible.
+
+## Decision
+
+Adopt an **OpenAPI-driven, spec-first release model** with **tag-based deployments** to align BE(s) and FE:
+
+### Core Principle: Deploy from Tags, Not from Main
+
+Both backend and frontend will **only deploy tagged releases** to shared environments (`dev`, `qa`, `prod` and `feature env`).
+This ensures:
+- Explicit version tracking for troubleshooting
+- Controlled, coordinated releases between teams
+- Ability to keep environments stable during critical testing periods
+
+### 1. **Spec-first PRs**
+   - Any API change updates `openapi.yaml` in the backend repo first.
+   - FE + BE reviewers required on spec-changing PRs.
+
+### 2. **Breaking-change gate (OpenAPI diff)**
+   - CI compares PR's `openapi.yaml` to the **last released tag's** spec; **breaking diffs fail** the PR.
+   - Large breaking changes (like API V2) **must not merge to `main`** until ready for general deployment:
+     - Use long-lived feature branches (e.g., `feat/api-v2`)
+     - Deploy feature branches to dedicated preview environments for testing
+     - Only merge to `main` when both BE and FE are ready to adopt the changes
+   - Smaller breaking changes must ship as **vN → vN+1** with coexistence of vN for one train,
+   or be adapted via a temporary shim at the gateway.
+
+### 3. **Explicit backend version pinning**
+   - Frontend explicitly tracks which backend tag/version it is compatible with (e.g., via environment config or documentation).
+   - FE upgrades to work with new backend versions are explicit and coordinated (not automatic).
+
+### 4. **Provider conformance tests**
+   - Backend CI runs **conformance tests** validating real responses against the OpenAPI schemas for
+   critical endpoints (e.g.,Schemathesis runs).
+   - Release promotion requires green conformance.
+
+### 5. **Environment strategy & promotions**
+   - **Shared environments (`dev`, `preprod`, `prod`)** only receive tagged releases:
```

Comment:
I'd recommend using tags instead of creating tagged releases to avoid cluttering the releases page on github